src/types.rs

```diff
@@ -581,4 +586,43 @@ mod test {
         let api_key = ApiKey("55555".to_string());
         assert!(format!("{api_key:?}") == "{API_KEY}");
     }
+
+    #[test]
+    fn test_bool_storable() {
```

Comment:
Updated to use the latest version. The caveat is that `candid::Principal` is distinct from `ic_principal::Principal` (which has the `Storable` implementation), requiring some conversions between the two. Overall, this is probably a more elegant approach regardless.

Reply:
I timeboxed trying to use the latest version of `ic-stable-structures`, and the issue that I ran into was that all of the latest versions which implement `Storable` for `Principal` and `bool` use Candid 10.x, which conflicts with transitive dependencies from the forked `cketh-minter` branch. 

After spending quite a lot of time trying to work around this (and running into more and more version conflicts), it might make sense to wait until after we can drop the `cketh-minter` dependency before making this change. However, I was able to remove `StringStorable` to simplify the codebase.

Reply:
> use Candid 10.x, which conflicts with transitive dependencies from the forked cketh-minter branch.

ha I see, another good reason to drop that dependency 😅.

> it might make sense to wait until after we can drop the cketh-minter dependency before making this change in another PR

Sure that sounds ok to me. I'm trying to also push on that front, since it will be needed when we want to implement something new (e.g. 3-out-of-4 strategy)